src/d/actor/d_a_obj_hat.cpp

```diff
@@ -4,91 +4,241 @@
 //
 
 #include "d/actor/d_a_obj_hat.h"
+#include "d/actor/d_a_npc_roten.h"
+#include "d/d_a_obj.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_kankyo.h"
 #include "d/d_procname.h"
+#include "d/res/res_ro.h"
+#include "JSystem/J3DGraphAnimator/J3DModel.h"
+
+static const int l_bmd_ix_tbl[] = {RO_BDL_RO_HAT, RO_BDL_RO_HAT2, RO_BDL_RO_HAT3, RO_BDL_RO_HAT};
+static const int l_bck_ix_tbl[] = {RO_INDEX_BCK_HAT_WID, RO_BCK_HAT2_WIND, RO_BCK_HAT3_WID,
+                                   RO_BCK_HAT_WID};
+static dCcD_SrcCyl l_cyl_src = {
+    // dCcD_SrcGObjInf
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ 0,//AT_TYPE_UNK1000,
+        /* SrcObjAt  Atp     */ 0,
+        /* SrcObjAt  SPrm    */ 0,
+        /* SrcObjTg  Type    */ AT_TYPE_WIND,
+        /* SrcObjTg  SPrm    */ cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+        /* SrcObjCo  SPrm    */ cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsOther_e | cCcD_CoSPrm_VsGrpAll_e,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,//AT_TYPE_WIND,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ dCcG_TgSPrm_NoHitMark_e | dCcG_TgSPrm_NoConHit_e,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGCylS
+    {
+        /* Center */ 0.0f,
+        0.0f,
+        0.0f,
+        /* Radius */ 70.0f,
+        /* Height */ 80.0f,
+    },
+};
+
+
 
 /* 00000078-00000184       .text __ct__10daObjHat_cFv */
 daObjHat_c::daObjHat_c() {
-    /* Nonmatching */
+    u32 hatno = getPrmHatNo();
+    mHatNo = hatno;
+    mState = 0;
 }
 
 /* 000003AC-000003CC       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* actor) {
+    return ((daObjHat_c*)actor)->createHeap();
 }
 
 /* 000003CC-0000045C       .text _create__10daObjHat_cFv */
 s32 daObjHat_c::_create() {
-    /* Nonmatching */
+    // if (this->actor_condition & )
+    fopAcM_SetupActor(this, daObjHat_c);
+
+    int var = dComIfG_resLoad(&mPhs, "Ro");
+
+    if (var == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap((fopAc_ac_c*)this, &CheckCreateHeap, 0uL)) {
```

Comment:
```suggestion
        if (!fopAcM_entrySolidHeap(this, &CheckCreateHeap, 0uL)) {
```
This cast shouldn't be necessary (unless you added it to fix regalloc)